pkg/lsp/service.go

```diff
@@ -200,9 +201,36 @@ type Hover struct {
 	Range    *Range         `json:"range,omitempty"`
 }
 
-type MarkedString struct {
+type MarkedString markedString
+
+type markedString struct {
```

Comment:
I don't understand the reason for the type alias? You should just be able to implement the marshallers on `type MarkedString struct`